src/main/java/com/yahoo/bullet/querying/Querier.java

```diff
@@ -390,40 +381,41 @@ public Querier(Mode mode, RunningQuery query, BulletConfig config) {
             rateLimit = new RateLimiter(maxEmit, timeInterval);
         }
 
-        Optional<List<BulletError>> errors;
+        Query query = runningQuery.getQuery();
 
-        errors = runningQuery.initialize();
-        if (errors.isPresent()) {
-            return errors;
+        if (query.getFilter() != null) {
+            filter = new Filter(query.getFilter());
         }
 
-        Query query = this.runningQuery.getQuery();
+        switch (query.getProjection().getType()) {
+            case COPY:
+                projection = new Projection(query.getProjection().getFields());
+                copy = true;
+                break;
+            case NO_COPY:
+                projection = new Projection(query.getProjection().getFields());
+                copy = false;
+                break;
+            default:
+                break;
+        }
 
         // Aggregation and Strategy are guaranteed to not be null.
-        Strategy strategy = AggregationOperations.findStrategy(query.getAggregation(), config);
-        errors = strategy.initialize();
-        if (errors.isPresent()) {
-            return errors;
-        }
+        Strategy strategy = query.getAggregation().getStrategy(config);
 
-        if (query.getPostAggregations() != null) {
-            postStrategies = query.getPostAggregations().stream().map(PostAggregationOperations::findPostStrategy).collect(Collectors.toList());
-            for (PostStrategy postStrategy : postStrategies) {
-                errors = postStrategy.initialize();
-                if (errors.isPresent()) {
-                    return errors;
-                }
-                addTransientFieldsFor(postStrategy);
-            }
+        List<PostAggregation> postAggregations = query.getPostAggregations();
+        if (postAggregations != null && !postAggregations.isEmpty()) {
+            postStrategies = postAggregations.stream().map(PostAggregation::getPostStrategy).collect(Collectors.toList());
         }
 
-        // Scheme is guaranteed to not be null.
-        window = WindowingOperations.findScheme(query, strategy, config);
-        return window.initialize();
+        // Scheme is guaranteed to not be null. It is constructed in its "start" state.
+        window = query.getWindow().getScheme(strategy, config);
+
+        runningQuery.start();
```

Comment:
Should the startTime come from the Query itself now? Might not need a RunningQuery anymore since it just has the timeout logic. That can go here.